File: 01-BruteForce-and-Backtracking/tldusqkr/C-7568.py
people_n = int(input())
people_info = list()
for i in range(people_n):
    weight, height = map(int, input().split(' '))
    people_info.append((weight, height))

for person in people_info:
    rank = 1

    for nextPerson in people_info:
        if (person[0] != nextPerson[0]) and (person[1] != nextPerson[1]):
            if(person[0] < nextPerson[0] and person[1] < nextPerson[1]):
                rank += 1

    print(rank, end=" ")

File: C-7568.cpp
#include <iostream>
using namespace std;

int main(){
    int *weight, *height, *rank;
    int arr_size;

    cin >> arr_size;

    weight = new int[arr_size];
    height = new int[arr_size];
    rank = new int[arr_size];

    for(int i = 0; i < arr_size; i++){
        cin >> weight[i] >> height[i];
    }

    for(int i = 0; i < arr_size-1; i++){
        for(int j = i; j < arr_size; j++){
            if( (weight[i] < weight[j]) && (height[i] < height[j]) ) rank[i]++;
            else if ((weight[i] > weight[j]) && (height[i] > height[j]) ) rank[j]++;
        }
    }

    for(int i = 0; i < arr_size; i++){
        cout << rank[i]+1 << " ";
    }
    delete[] weight;
    delete[] height;
    delete[] rank;

    return 0;
}